core/src/operations/decorators/advice.rs

```diff
@@ -1,86 +1,189 @@
     ///
-    /// The operand stack is expected to be arranged as follows (from the top):
-    /// - key, 4 elements.
-    /// - root of the Sparse Merkle tree, 4 elements.
+    /// Inputs:
+    ///   Operand stack: [KEY, ROOT, ...]
+    ///   Advice stack: [...]
+    ///
+    /// Outputs:
+    ///   Operand stack: [KEY, ROOT, ...]
+    ///   Advice stack: [f0, f1, K, V, f2]
     ///
-    /// After a successful operation, the advice stack will look as follows:
-    /// - boolean flag set to `1` if the depth is `16` or `48`.
-    /// - boolean flag set to `1` if the depth is `16` or `32`.
-    /// - remaining key word; will be zeroed if the tree don't contain a mapped value for the key.
-    /// - value word; will be zeroed if the tree don't contain a mapped value for the key.
-    /// - boolean flag set to `1` if a remaining key is not zero.
+    /// Where:
+    /// - f0 is a boolean flag set to `1` if the depth is `16` or `48`.
+    /// - f1 is a boolean flag set to `1` if the depth is `16` or `32`.
+    /// - K is the remaining key word; will be zeroed if the tree don't contain a mapped value
+    ///   for the key.
+    /// - V is the value word; will be zeroed if the tree don't contain a mapped value for the key.
+    /// - f2 is a boolean flag set to `1` if a remaining key is not zero.
     SmtGet,
+
+    // ADVICE MAP INJECTORS
+    // --------------------------------------------------------------------------------------------
+    /// Reads words from memory at the specified range and inserts them into the advice map under
+    /// the key `KEY` located at the top of the stack.
+    ///
+    /// Inputs:
+    ///   Operand stack: [KEY, start_addr, end_addr, ...]
+    ///   Advice map: {...}
+    ///
+    /// Outputs:
+    ///   Operand stack: [KEY, start_addr, end_addr, ...]
+    ///   Advice map: {KEY: values}
+    ///
+    /// Where `values` are the elements located in memory[start_addr..end_addr].
+    MemToMap,
+
+    /// Reads two word from the operand stack and inserts them into the advice map under the key
+    /// defined by the hash of these words.
+    ///
+    /// Inputs:
+    ///   Operand stack: [B, A, ...]
+    ///   Advice map: {...}
+    ///
+    /// Outputs:
+    ///   Operand stack: [B, A, ...]
+    ///   Advice map: {KEY: [a0, a1, a2, a3, b0, b1, b2, b3]}
+    ///
+    /// Where KEY is computed as hash(A || B, domain), where domain is provided via the immediate
+    /// value.
+    HdwordToMap { domain: Felt },
 }
 
 impl fmt::Display for AdviceInjector {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         match self {
-            Self::MerkleNode => write!(f, "merkle_node"),
-            Self::MerkleMerge => write!(f, "merkle_merge"),
-            Self::DivResultU64 => write!(f, "div_result_u64"),
-            Self::MapValue => write!(f, "map_value"),
-            Self::Memory => write!(f, "mem"),
+            Self::MerkleNodeMerge => write!(f, "merkle_tree_merge"),
```

Comment:
Should we also update the display to be `merkle_node_merge`?

Reply:
Good catch! Updated.